### ISSUE TITLE
A11y: structure of article overviews could be improved

### ISSUE THREAD

**Issue by dstorey** [smashingmagazine/redesign#156]
https://next.smashingmagazine.com/articles/

Inspect article previews.

One issue is tab order (and perhaps reading order in screen reader) doesn't follow document order and document semantics. This could be a WCAG 1.3.2 failure, although not super bad here as the previews are short.

Normally you read and tab through the header, main content, then footer, and your eyes move from left to right, top to bottom (for roman languages). With this layout, the footer is in the right hand column, but placed before the main content in the DOM (and thus tab order)

See screenshot of screen reader reading order and tab order (red and blue respectively...sorry about the finger drawing, I left my pen at work)
![sketch 1](https://cloud.githubusercontent.com/assets/465069/24137118/8d652b08-0dcf-11e7-8768-a2072ab87f97.png)

This is easy to fix by moving the footer element below the content (last element in the article element).  
this works without any style fixes and makes the document structure correct too (footer at the foot of the article).

The next issue may be a bit more tricky as it requires CSS changes.

Currently there is a header element inside the article (so far so good), but the h2 heading is outside the heading. This makes an odd document outline. It should be inside (note browsers don't need to report headers and footers inside articles to the a11y layer per html aam spec usually just relying on the h* element, but your mileage may vary with different browser/sr combinations.

The next thing that is off is that inside the heading is a figure element (without any figcaption so it has no accessible name; see https://w3c.github.io/html-aam/#h-figure-element-accessible-name-computation). 

In this case I don't think the author portrait is a figure of the article. It is really decoration, as the author name is visibly along side it (leading to the name read twice, which was reported as another issue). Instead it would be better to remove the figure and just have `img alt=""`. You probably want it as a link, but you can just include it in the same anchor where the author name is included.

The whole img, author name and "wrote" span could be a paragraph above the h2 inside the header element, or arguably as it is meant to be all read together as one sentence, they could be a span inside the h2 (depends if you want that as one span of text as the accessible name of the article heading and if you want it read when the user navigates by headings on the page or not.

**Comment by indysigner** [comment]
Great findings, thank you @dstorey! Moving this issue into our private repository in a moment.

**Comment by indysigner** [comment]
This issue was moved to smashingmagazine/smashing-magazine#578
